res/res_rtp_asterisk.c

```diff
@@ -1914,6 +1913,27 @@ struct dtls_cert_info {
 	X509 *certificate;
 };
 
+static int apply_dh_params(SSL_CTX *ctx, BIO *bio)
+{
+	int res = 0;
+
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	EVP_PKEY *dhpkey = PEM_read_bio_Parameters(bio, NULL);
+	if (dhpkey) {
+		res = EVP_PKEY_is_a(dhpkey, "DH") && SSL_CTX_set0_tmp_dh_pkey(ctx, dhpkey);
+	}
+	EVP_PKEY_free(dhpkey);
+#else
+	DH *dh = PEM_read_bio_DHparams(bio, NULL, NULL, NULL);
+	if (dh) {
+		res = SSL_CTX_set_tmp_dh(ctx, dh);
+	}
+	DH_free(dh);
```

Comment:
DH_free should be inside if (dh) {}

Reply:
[From `man 3 DH_free`](https://www.openssl.org/docs/man3.0/man3/DH_free.html):

> DH_free() frees the DH structure and its components. The values are erased before the memory is returned to the system.  If dh is NULL nothing is done.